osquery/tables/utility/file.cpp

```diff
@@ -25,8 +32,144 @@ namespace osquery {
 
 namespace tables {
 
-#if !defined(WIN32)
+namespace {
+#ifdef WIN32
+
+struct LnkData {
+  std::string target_path;
+  std::string target_type;
+  std::string target_location;
+  std::string start_in;
+  std::string run;
+  std::string comment;
+};
+
+std::string showCmdToString(int show_cmd) {
+  switch (show_cmd) {
+  case SW_SHOWNORMAL: {
+    return "Normal window";
+  }
+  case SW_SHOWMAXIMIZED: {
+    return "Maximized";
+  }
+  case SW_SHOWMINIMIZED: {
+    return "Minimized";
+  }
+  default: {
+    return "Unknown";
+  }
+  }
+}
+
+boost::optional<LnkData> parseLnkData(const fs::path& link) {
+  IShellLink* shell_link;
+  auto hres = CoCreateInstance(CLSID_ShellLink,
+                               nullptr,
+                               CLSCTX_INPROC_SERVER,
+                               IID_IShellLink,
+                               reinterpret_cast<LPVOID*>(&shell_link));
+  if (FAILED(hres)) {
+    TLOG << "Failed to create an instance of a shell link, error: "
+         << shell_link;
+    return boost::none;
+  }
+
+  IPersistFile* file;
+  hres = shell_link->QueryInterface(IID_IPersistFile,
+                                    reinterpret_cast<LPVOID*>(&file));
```

Comment:
I think `file` needs to be freed with a `file->Release()`? Shall we use a `scope_guard` here?

Reply:
🤦 You're correct, thanks!